tests/callbacks/test_image_visualizer.py

```diff
@@ -1,112 +1,235 @@
 # Copyright 2022 MosaicML Composer authors
 # SPDX-License-Identifier: Apache-2.0
 
+import imghdr
+import json
 import os
+import zipfile
 from pathlib import Path
 
 import pytest
 from torch.utils.data import DataLoader
 
 from composer.callbacks import ImageVisualizer
-from composer.core import Time, TimeUnit
-from composer.loggers import InMemoryLogger, WandBLogger
+from composer.loggers import WandBLogger
+from composer.loggers.logger import Logger
 from composer.trainer import Trainer
-from tests.common.datasets import RandomImageDataset
-from tests.common.models import SimpleConvModel
-
-try:
-    import wandb
-    _WANDB_INSTALLED = True
-    del wandb  # unused
-except ImportError:
-    _WANDB_INSTALLED = False
-
-
-@pytest.mark.skipif(not _WANDB_INSTALLED, reason='Wandb is optional')
-@pytest.mark.parametrize('interval', ['9ba', '90ba', '2ep', '3ep', '7ep'])
-def test_image_visualizer(interval: str):
-    # Construct the callback
-    image_visualizer = ImageVisualizer(interval=interval)
-    in_memory_logger = InMemoryLogger()  # track the logged images in the in_memory_logger
-
-    num_train_epochs = 9
-
-    # Construct the trainer and train
-    trainer = Trainer(
-        model=SimpleConvModel(),
-        callbacks=image_visualizer,
-        loggers=in_memory_logger,
-        train_dataloader=DataLoader(RandomImageDataset()),
-        eval_dataloader=DataLoader(RandomImageDataset()),
-        max_duration=f'{num_train_epochs}ep',
-    )
+from tests.common.datasets import RandomImageDataset, RandomSegmentationDataset
+from tests.common.models import SimpleConvModel, SimpleSegmentationModel
+
+
+@pytest.fixture
+def test_wandb_logger(tmp_path, dummy_state):
+    pytest.importorskip('wandb', reason='wandb is optional')
+    os.environ['WANDB_DIR'] = str(tmp_path)
```

Comment:
I'm not sure about env var setting in a specific test fixture, because this sets them for all tests, which may cause unexpected results across tests. I'd rather either add whatever WandB env vars you want to the autoused fixture, or unset these env vars at the end of the test. Perhaps a minor thing, not sure what you think.

Reply:
I'm ok with this if you think its ok